src/Init/Data/List/Nat/Basic.lean

```diff
@@ -109,6 +109,42 @@ theorem length_rightpad {n : Nat} {a : α} {l : List α} :
   simp [rightpad]
   omega
 
+/-! ### intersperse -/
+
+@[simp] theorem length_intersperse (l : List α) (sep : α) :
+    (l.intersperse sep).length = 2 * l.length - 1 := by
+  fun_induction intersperse <;> simp only [intersperse, length_cons, length_nil] at *
+  rename_i h _
+  have := length_pos_iff.mpr h
+  omega
+
+@[simp] theorem getElem?_intersperse_two_mul {l : List α} (sep : α) :
+    (l.intersperse sep)[2 * i]? = l[i]? := by
+  induction l using intersperse.induct_unfolding sep generalizing i <;> cases i
+  all_goals simp [mul_succ, *]
+
+theorem getElem?_intersperse_two_mul_add_one {l : List α} (h : i + 1 < l.length) (sep : α) :
+    (l.intersperse sep)[2 * i + 1]? = some sep := by
+  fun_induction intersperse generalizing i
+  · contradiction
+  · contradiction
+  · rename_i hn _
+    have ⟨_, tl, _⟩ := ne_nil_iff_exists_cons.mp hn
+    cases tl <;> cases i <;> simp_all +arith
+
+@[simp] theorem getElem_intersperse_two_mul
+    {l : List α} (h : 2 * i < (l.intersperse sep).length) :
```

Comment:
It's a bit strange that `sep` is an autoimplicit in the last two lemmas but not in the others.